### PR TITLE
[JBPM-9786] Add REST Endpoint for claiming tasks in bulk

### PR DESCRIPTION
**Pull request by fjtirado**
Adding query param and test

**JIRA**: 

[link](https://github.com/fjtirado/droolsjbpm-integration/pull/new/JBPM-9786)

**referenced Pull Requests**:
https://github.com/kiegroup/jbpm/pull/1955

### CODE REVIEW

**Comment by fjtirado** [comment]
jenkins retest this

**Review by elguardian — APPROVED** [review]
LGTM

**Comment by fjtirado** [comment]
jenkins retest this

File: kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/UserTaskServiceBase.java [hunk 1]
```diff
@@ -85,11 +86,15 @@ public void activate(String containerId, Number taskId, String userId) {
...
     public void claim(String containerId, Number taskId, String userId) {
-
         userId = getUser(userId);
         logger.debug("About to claim task with id '{}' as user '{}'", taskId, userId);
-        userTaskService.claim(taskId.longValue(), userId);
-
+        userTaskService.claim(containerId, taskId.longValue(), userId);
+    }
+    
+    public void claim(String containerId, Collection<Long> taskIds, String userId) {
```
Comment (elguardian) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/UserTaskServiceBase.java:94]: @fjtirado failures related this happens you need in the service base to set as Number not as Long

File: kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/UserTaskServicesClientImpl.java [hunk 2]
```diff
@@ -132,6 +132,20 @@ public void claimTask(String containerId, Long taskId, String userId) {
...
         }
     }
+    
+    @Override
+    public void claimTask(String containerId, List<Long> taskIds, String userId) {
+        if( config.isRest() ) {
+            String queryString = buildQueryString("taskId",taskIds);
+            queryString+= getUserQueryStr(userId);
+            makeHttpPutRequestAndCreateCustomResponse(
+                    build(loadBalancer.getUrl(), TASK_URI + "/" + TASK_INSTANCE_CLAIM_PUT_URI, Collections.singletonMap(CONTAINER_ID, containerId)) + queryString, null, String.class, getHeaders(null));
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/UserTaskServicesClientImpl.java:142]: Shouldn't it be TASK_INSTANCE_CLAIM_BULK_URI?
```suggestion
                    build(loadBalancer.getUrl(), TASK_URI + "/" + TASK_INSTANCE_CLAIM_BULK_URI, Collections.singletonMap(CONTAINER_ID, containerId)) + queryString, null, String.class, getHeaders(null));
```

**Review by gmunozfe — CHANGES_REQUESTED** [review]
Check URI at client method

File: kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/UserTaskServicesClientImpl.java (hunk 2 above)
Comment (fjtirado) [reply kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/UserTaskServicesClientImpl.java]: Thanks!!!

File: kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/UserTaskServiceBase.java (hunk 1 above)
Comment (fjtirado) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/UserTaskServiceBase.java:94]: I think the failure is caused by the typo detected by Gonzalo https://github.com/kiegroup/droolsjbpm-integration/pull/2518#discussion_r655945914
Lets wait this round to see,

**Review by gmunozfe — APPROVED** [review]
Looks good to me, amazing work @fjtirado !

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java [hunk 3]
```diff
@@ -247,6 +248,53 @@ public void testReleaseAndClaim() throws Exception {
...
+            taskClient.releaseTask(CONTAINER_ID, taskSummary.getId(), USER_YODA);
+
+            taskList = taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10);
+            assertNotNull(taskList);
+
+            assertEquals(1, taskList.size());
+            taskSummary = taskList.get(0);
+            checkTaskNameAndStatus(taskSummary, "First task", Status.Ready);
+
+            taskClient.claimTask(CONTAINER_ID, Collections.singletonList(taskSummary.getId()), USER_YODA);
```
Comment (afalhambra) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java:276]: Maybe I would try to claim more than 1 task to make sure we're reserving all the tasks passed as id, or combining with a non-claimable task to test it still works when one task cannot be reserved.

**Review by afalhambra — APPROVED** [review]
great job, just two optional comments.

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java (hunk 3 above)
Comment (fjtirado) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskServiceIntegrationTest.java:276]: Im not really sure this "implementation detail" should be checked by an existing test. 
As I said in other forum, I have serious doubts current silent approach (we are not communicating client the errors we are catching for individual operations) is the way to go. I implemented that way to no do differently to other bulk operations, but I prefer not enforce that in a test

File: kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/UserTaskServicesClient.java [hunk 4]
```diff
@@ -32,6 +32,8 @@
     void activateTask(String containerId, Long taskId, String userId);
 
     void claimTask(String containerId, Long taskId, String userId);
+    
+    void claimTask(String containerId, List<Long> taskIds, String userId);
```
[the remark this answers is not in the document]
Comment (fjtirado) [reply kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/UserTaskServicesClient.java]: Thaks ok ;), changed.

**Comment by fjtirado** [comment]
jenkins retest this

**Comment by fjtirado** [comment]
jenkins retest this

**Comment by fjtirado** [comment]
jenkins test this

**Comment by fjtirado** [comment]
please jenkins test this

**Comment by fjtirado** [comment]
jenkins test this

**Comment by fjtirado** [comment]
jenkins retest this

**Pull request merged**